static/src/javascripts/projects/common/modules/email/email.js

```diff
@@ -226,12 +226,13 @@ const setIframeHeight = (
     iFrameEl: HTMLIFrameElement,
     callback: () => void
 ): (() => void) => () => {
-    fastdom.write(() => {
-        iFrameEl.height = '';
-        iFrameEl.height = `${
-            iFrameEl.contentWindow.document.body.clientHeight
-        }px`;
-        callback();
+    fastdom.read(() => {
+        const height = iFrameEl.contentWindow.document.body.clientHeight;
+
+        fastdom.write(() => {
```

Comment:
Should we avoid calling `write` (mutate queue) from inside a `write` (measure queue)? In other places, we handle this like so:

```js
fastdom.read(() => iFrameEl.contentWindow.document.body.clientHeight)
  .then(height => {
    fastdom.write(() => {
      iFrameEl.height = `${height}px`;
      callback();
    });
  });
```

Can't find any specific docs on this!

Reply:
I think this is okay from the initial fastdom blog post and other examples:

![screenshot 2019-01-31 at 15 49 35](https://user-images.githubusercontent.com/858402/52066142-dcf2ad80-256f-11e9-8451-2f799aaafda9.png)

Having said that, I don't know how deep the rabbit hole goes - i.e. can you just nest forever?!

Reply:
Ah, actually we'd need to swap out `fastdom` for [`lib/fastdom-promise`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/lib/fastdom-promise.js) in this module

Reply:
hey, so made this change. I've looked at https://github.com/wilsonpage/fastdom/blob/master/fastdom.js#L75 though and can't see any expensive operations (it adds stuff to an array, and does a few conditionals and schedules a raf callback) so suspect this is unlikely to be significant in practice.

Reply:
Yeah, I think you're right, it shouldn't make any difference